src/services/db/GitFileSystemService.ts

```diff
@@ -203,6 +203,28 @@ export default class GitFileSystemService {
       })
   }
 
+  // Determine if the given branch is present in the repo
+  isBranchPresent(
+    repoName: string,
+    branchName: string
+  ): ResultAsync<boolean, never> {
+    const efsVolPath = this.getEfsVolPathFromBranch(branchName)
+    return ResultAsync.fromPromise(
+      this.git
+        .cwd({ path: `${efsVolPath}/${repoName}`, root: false })
+        .revparse(["--verify", branchName]),
```

Comment:
can consider using `.branchLocal` instead

Reply:
also regarding what you mentioned just now about making it not return an error, I think as a common util method, this should still return an error. The caller should be the one to decide if they want to ignore the error i.e `orElse(()=> false)` or something along those lines